src/compiler/parser.ts

```diff
@@ -8036,6 +8036,14 @@ namespace ts {
                             case SyntaxKind.EndOfFileToken:
                                 // Done
                                 break loop;
+                            case SyntaxKind.MinusToken:

```

Comment:
I don't think `-` should be skipped in the parser, since it's not parsed in any other way. The handling of `*` and indentation is bad enough and I don't want to expand that.

I also think it should only be skipped right after a param name or other name. In other contexts it should be retained. I'd recommend post-processing the comment similar to your removal of `|` in `@link`.